internal/services/environment/api_environment.go

```diff
@@ -527,6 +541,19 @@ func (client *Client) UpdateEnvironmentAiFeatures(ctx context.Context, environme
 	return nil
 }
 
+func (client *Client) handleHttpConflict(ctx context.Context, apiResponse *api.Response) error {
+	body := string(apiResponse.BodyAsBytes)
+	if body == "" {
+		return errors.New("environment failed with HTTP 409. No body in response")
+	}
+	// if 409 returns anything other than another ongoing lifecycle operation, fail the request and return the body as error to the user
+	if !strings.Contains(body, "OperationNotStartable") {
+		return errors.New("environment failed with HTTP 409. Body: " + body)
+	}
+	tflog.Debug(ctx, "Another lifecycle operation is in progress, waiting for it to complete")
+	return client.Api.SleepWithContext(ctx, api.DefaultRetryAfter())
```

Comment:
use the retry function that uses the headers if possible.

Reply:
That functinality is later in the code. This method handles 409's on requests that **should** return lifecycle operation response with location header but they haven't because they returned 409. In case of other operation running the L554 waits for it to finish and then the code flow continues with a correct response that includes location header etc.